src/globus_sdk/exc/warnings.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+import os
+import typing as t
+import warnings
+
+
+# this is (by meaning) a type of DeprecationWarning, but because python suppresses
+# DeprecationWarning by default, most users would never see it if we inherited from that
+# therefore, inheriting from Warning is an "incorrect" but pragmatic choice
+#
+# in concert with the helpers below to make the warnings opt-in, this results in good
+# user-facing behavior
+class GlobusSDKLegacyBehaviorWarning(Warning):
```

Comment:
Did you intentionally not name it `GlobusSdkDeprecationWarning` since we don't extend `DeprecationWarning`?

`Deprecation` more accurately states "we're taking this thing away" than `LegacyBehavior` imo

Reply:
It is intentional, and you guessed right about the naming, but both you and Kurt have raised concerns about how it plays out.
I plan for us to discuss this at the SDK meeting today, so we don't have to get too deep into it right now.

The one thing which I think is useful to know ahead of time is that the interpreter has a default filter for warnings which suppresses `DeprecationWarning` (and subclasses thereof). So you will only see a `DeprecationWarning` if you explicitly adjust the warning filters or use a program like `pytest` which sets warning filters for you.